kong/db/schema/typedefs.lua

```diff
@@ -63,15 +66,34 @@ local function validate_ip_or_cidr(ip)
 end
 
 
-local function validate_cidr_v4(ip)
-  local _, err = iputils.parse_cidr(ip)
+local function validate_cidr_v4(ip_or_cidr)
+  local is_ipv4 = ipmatcher.parse_ipv4(ip_or_cidr)
+  if is_ipv4 then
+    return true
+  end
 
-  -- It's an error only if the second variable is a string
-  if type(err) == "string" then
-    return nil, "invalid cidr range: " .. err
+  local p = find(ip_or_cidr, "/", 1, true)
+  if not p then
+    return nil, "invalid ipv4 cidr range:: '" .. ip_or_cidr .. "'"
   end
 
-  return true
+  local ip = sub(ip_or_cidr, 1, p - 1)
+  if not ipmatcher.parse_ipv4(ip) then
+    return nil, "invalid ipv4 cidr range: " .. ip_or_cidr
+  end
+
+  local block = tonumber(sub(ip_or_cidr, p + 1), 10)
+  if not block then
+    return nil, "invalid ipv4 cidr range: " .. ip_or_cidr
+  end
+
+  for i = 0, 32 do
+    if block == i then
+      return true
+    end
+  end
```

Comment:
How about:
```
return block >= 0 and block <=32
```

instead? Doesn't seems that the loop is necessary.

Reply:
Actually, if we are concerned about people feeding CIDR like this:

`127.0.0.0/12.5`

we could just check for whether a fraction part exists. So:

```
if math.floor(block) ~= block then
  return nil, "invalid ipv4 cidr range: " .. ip_or_cidr .. " prefix length must be a whole number"
end
```

should allow us to get rid of the loop still.

Reply:
Also `0xa` is probably passing.

Reply:
I was also thinking about building a lookup table.

Reply:
It is now changed to `lookup table`.

Reply:
I think @fffonion recommended to use a regex to check for the CIDR format, might be easier and more robust.

Reply:
This is the most robust I can think of. Won't get anymore robust with regex. Just inserts complex tech to simple problem already solved.